metricflow/test/integration/test_cases/itest_semi_additive_measure.yaml

```diff
@@ -53,4 +53,51 @@ integration_test:
     ON
       a.ds = b.ds
     GROUP BY
-      a.ds
\ No newline at end of file
+      a.ds
+---
+integration_test:
```

Comment:
Is there a more minimal way to evaluate this condition, like maybe a test case in test_measure_additiveness.py? I think that'll be easier to understand (although having an integration test exercise this join path is a good idea!).

If you can come up with a more focused test, please add that as well and then you can update the docstring below to be a bit broader, more focused on ensuring correct results with a join condition.

Reply:
I can't think of a more minimal way of evaluating this. In the bigger picture, this is generally a problem where in some instance we want to copy a `xSpec` (in this case `MeasureSpec`) and in the code it ended up just doing
```python
XSpec(element_name="...")
```
even though that spec has more attributes and should pass along the rest of the attributes. So this technically became a problem because we added a new attribute to `MeasureSpec`, which ended up unraveling this error. So whoever adds a new attribute (in this case, me) shouldve checked that all the constructors were passed correctly! Or we should think of a better way of copying these (maybe with `deepcopy`?) but these are immutable anyways so maybe we can just copy directly.

Reply:
We should definitely find a better way to copy these things around, that's a known issue.